src/nodes/composite/predicate.ts

```diff
@@ -264,12 +278,38 @@ export const assertAllowsConstraint = (
         case "morph":
             return
         default:
-            throwInternalError(`Unexpxected rule kind '${kind}'`)
+            throwInternalError(`Unexpxected rule kind '${node.kind}'`)
+    }
+}
+
+const assertValidLimit = (bounds: Range, boundType: "number" | "Date") => {
+    for (const index in bounds) {
+        const boundKind: BoundKind = index === "0" ? "right" : "left"
+        if (boundType === "number") {
+            if (typeof bounds[index].limit !== boundType) {
+                throwParseError(
```

Comment:
This is the same as the error on `300`, so it can be combined into a single if check (if it gets hard to read you could define a couple intermediate variables to store the conditions like `isNumberMismatch` and `isDateMismatch`).